src/Payments.sol

```diff
@@ -1113,11 +1203,46 @@ contract Payments is
             "failed to settle: insufficient lockup to cover settlement"
         );
 
-        // Transfer funds from payer to payee
+        // Transfer funds from payer (always pays full settled amount)
         payer.funds -= settledAmount;
-        payee.funds += settledAmount;
 
-        // Reduce the lockup by the settled amount
+        // Calculate payment contract fee (if any) based on full settled amount
+        paymentFee = 0;
+        if (PAYMENT_COMISSION_BPS > 0) {
+            paymentFee = (settledAmount * PAYMENT_COMISSION_BPS) /
+                COMMISSION_MAX_BPS;
+        }
+
+        // Calculate amount remaining after contract fee
+        uint256 amountAfterPaymentFee = settledAmount - paymentFee;
+
+        // Calculate operator commission (if any) based on remaining amount
+        operatorCommission = 0;
+        if (rail.commissionRateBps > 0) {
+            operatorCommission = (amountAfterPaymentFee * rail.commissionRateBps) /
+                COMMISSION_MAX_BPS;
+        }
```

Comment:
I like this approach but it makes it harder to reason about because I can't just "add the fees" together to figure out the overhead. I'd:

1. Take the fee as a percentage of the original amount settled.
2. Limit the max commission as 100% - payment contract fee.

But I don't feel too strongly about this.